modules/globebrowsing/src/globerotation.cpp

```diff
@@ -0,0 +1,236 @@
+#include <ghoul/logging/logmanager.h>
+#include <glm/gtx/quaternion.hpp>
+
+namespace {
+    constexpr openspace::properties::Property::PropertyInfo GlobeInfo = {
+        "Globe",
+        "Attached Globe",
+        "The globe on which the longitude/latitude is specified"
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo LatitudeInfo = {
+        "Latitude",
+        "Latitude",
+        "The latitude of the location on the globe's surface. The value can range from "
+        "-90 to 90, with negative values representing the southern hemisphere of the "
+        "globe. The default value is 0.0."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo LongitudeInfo = {
+        "Longitude",
+        "Longitude",
+        "The longitude of the location on the globe's surface. The value can range from "
+        "-180 to 180, with negative values representing the western hemisphere of the "
+        "globe. The default value is 0.0."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo AngleInfo = {
+        "Angle",
+        "Angle",
+        "A rotation angle that can be used to rotate the object around its own y-axis, "
+        "which will be pointing out of the globe's surface."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo UseHeightmapInfo = {
+        "UseHeightmap",
+        "Use Heightmap",
+        "If set to true, the heightmap will be used when computing the surface normal. "
+        "This means that the object will be rotated to lay flat on the surface at the "
+        "given coordinate and follow the shape of the landscape."
+    };
+
+    struct [[codegen::Dictionary(GlobeRotation)]] Parameters {
+        // [[codegen::verbatim(GlobeInfo.description)]]
+        std::string globe
+            [[codegen::annotation("A valid scene graph node with a RenderableGlobe")]];
+
+        // [[codegen::verbatim(LatitudeInfo.description)]]
+        std::optional<double> latitude;
+
+        // [[codegen::verbatim(LongitudeInfo.description)]]
+        std::optional<double> longitude;
+
+        // [[codegen::verbatim(AngleInfo.description)]]
+        std::optional<double> angle;
+
+        // [[codegen::verbatim(UseHeightmapInfo.description)]]
+        std::optional<bool> useHeightmap;
+    };
+#include "globerotation_codegen.cpp"
+} // namespace
+
+namespace openspace::globebrowsing {
+
+documentation::Documentation GlobeRotation::Documentation() {
+    return codegen::doc<Parameters>("globebrowsing_rotation_globerotation");
+}
+
+GlobeRotation::GlobeRotation(const ghoul::Dictionary& dictionary)
+    : _globe(GlobeInfo)
+    , _latitude(LatitudeInfo, 0.0, -90.0, 90.0)
+    , _longitude(LongitudeInfo, 0.0, -180.0, 180.0)
+    , _angle(AngleInfo, 0.0, 0.0, 360.0)
+    , _useHeightmap(UseHeightmapInfo, false)
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+    _globe = p.globe;
+    _globe.onChange([this]() {
```

Comment:
Is it supposed to be possible to change the globe during runtime? If not, then the `onChange` isn't really needed right? If yes, then I think the property should be added to the GUI, which it isn't at the moment.

Reply:
You can't change it in the GUI, because the property isn't added. But anyone could change it during runtime using the set property value functions

Reply:
Yea, I'd also say it's better not to allow re-globing.. since you, at the same time, probably also would want to reparent this object to a different scenegraph node as well, which currently does not work anyway

Reply:
Good point. Then I'll do the same change for the GlobeTranslation